### DISCUSSION TITLE
Fief admin keeps trying to run the "create your workspace" wizard if a non-admin user session cookie is present

### DISCUSSION THREAD

**Discussion by shrike71** [fief-dev/fief#16] [Q&A]
Just tried to log in as a non-admin user into the fief admin UI at:

http://<mysite>/admin/

The user interface keeps trying to run the "create your workspace" wizard if it detects a non-admin session cookie. Is this intended? Surely it should do something like re-route to the login page for non-privileged users (or reflect an error for the attempt)?

I get the same effect if i have generated a user session from the python client example generator and pasted in the url to complete the Oauth dance, too.

If i manually delete all the cookies, it'll prompt me for a login again and allow me to log in as an admin.

**Comment by frankie567** [comment]
Hello @shrike71!

Does your "regular user" registered in the *main* workspace, like your admin user?

If it is, the problem comes from the fact that currently, any user registered to the main workspace can actually access the admin dashboard and create new workspaces. It works well for our cloud version, but I understand it's not really ideal for self-hosted.

The current solution is to create a dedicated workspace for your application and keep the main one to manage Fief.

I've to think about a better way to handle this, maybe a parameter allowing access to the admin dashboard for new users. I would happily read your suggestions about this 🙂

**Reply by shrike71** [reply]
Hi frankie567,

That's absolutely fine... moved test users to their own workspace. I concur with some sort of flag disabling new users from registering with the main workspace (i.e. an existing admin will need to add you in).

Thanks again
